Ensure name columns are not cut off on APM tables

smith opened the pull request:
* Switch the overview to single column when we're at < 1200px
* Hide sparkplots on transactions, dependencies, and instances tables at widths where they could be easily cut off
* Right-align sparkplot columns
* Make chart heights match latency chart height when showing a single column
* Make all impact bars fixed 96px width
* Make values be shown before sparklines
* Make SparkPlots `overflow: 'hidden'`
* Remove fixed widths from columns
* Return breakpoint and width from `useBreakpoints`
* Rename "breakPoint" to "breakpoint" everywhere to match what EUI does
* Replace `ServiceListMetric` (on used on service list) with `ListMetric` (used when sparkplots are conditionally shown)

Here come the screenshots...

# Service inventory

## < 1200px

| Before | After |
| :---: | :---: |
|<img width="1213" alt="CleanShot 2021-09-07 at 13 13 11@2x" src="https://user-images.githubusercontent.com/9912/132391637-44c90414-fbbc-40e4-8bef-0a7e4b790807.png">|<img width="1209" alt="CleanShot 2021-09-07 at 13 12 49@2x" src="https://user-images.githubusercontent.com/9912/132391580-7bf60e49-a2a0-40e0-b359-d39d86b89e55.png">|

## ≥ 1200px

| Before | After |
| :---: | :---: |
|<img width="1209" alt="CleanShot 2021-09-07 at 13 13 51@2x" src="https://user-images.githubusercontent.com/9912/132391711-4fb3eb8b-57b1-45dd-a654-2359302f9842.png">|<img width="1210" alt="CleanShot 2021-09-07 at 13 14 09@2x" src="https://user-images.githubusercontent.com/9912/132391747-dbdcb416-f5cf-4fca-acd7-36537d505cb9.png">|

# Transactions overview

| Before | After |
| :---: | :---: |
|<img width="713" alt="CleanShot 2021-09-07 at 13 17 13@2x" src="https://user-images.githubusercontent.com/9912/132392183-28fc9e49-b1f3-43b3-8b7c-0449b2255fcc.png">|<img width="713" alt="CleanShot 2021-09-07 at 13 16 43@2x" src="https://user-images.githubusercontent.com/9912/132392128-2a561177-ccda-452b-ba3e-2ade09af859d.png">|

# Dependencies overview

| Before | After |
| :---: | :---: |
|<img width="723" alt="CleanShot 2021-09-07 at 13 18 39@2x" src="https://user-images.githubusercontent.com/9912/132392340-6bf1f3de-5ca6-447f-b1ab-1f5a09e79761.png">|<img width="722" alt="CleanShot 2021-09-07 at 13 18 58@2x" src="https://user-images.githubusercontent.com/9912/132392373-ed8d5eec-4df6-41cb-9b0d-cd64aad96f1e.png">|

# Service overview

## Transactions

(dependencies section behaves in the same way)

### 975px to 1200px

| Before | After |
| :---: | :---: |
|<img width="897" alt="CleanShot 2021-09-07 at 13 22 05@2x" src="https://user-images.githubusercontent.com/9912/132392767-ebbfe8be-1e3e-4726-a400-6f77614a1dc0.png">|<img width="896" alt="CleanShot 2021-09-07 at 13 22 24@2x" src="https://user-images.githubusercontent.com/9912/132392801-4c8dbc07-3766-44f7-b820-ea1705c0da85.png">|

### 1200px to 1600px

| Before | After |
| :---: | :---: |
| <img width="943" alt="CleanShot 2021-09-07 at 13 23 48@2x" src="https://user-images.githubusercontent.com/9912/132392979-4387c603-d656-492b-99c6-48be318268d6.png">|<img width="948" alt="CleanShot 2021-09-07 at 13 24 07@2x" src="https://user-images.githubusercontent.com/9912/132393013-9268f0ea-8eb8-4776-bfc8-3ef9257a9d21.png">|

### > 1600px

| Before | After |
| :---: | :---: |
|<img width="1348" alt="CleanShot 2021-09-07 at 13 25 52@2x" src="https://user-images.githubusercontent.com/9912/132393170-4b8e9b56-f2c4-4679-8d8a-578dcbbe7849.png">|<img width="1341" alt="CleanShot 2021-09-07 at 13 26 12@2x" src="https://user-images.githubusercontent.com/9912/132393217-9b613160-0521-48c6-9cc7-5f3a730a790b.png">|

## Instances table

### < 1600px

| Before | After |
| :---: | :---: |
|<img width="908" alt="CleanShot 2021-09-07 at 13 29 05@2x" src="https://user-images.githubusercontent.com/9912/132393536-d9b05a9b-63e2-41f9-aefa-f8bc0c365f98.png">|<img width="902" alt="CleanShot 2021-09-07 at 13 29 19@2x" src="https://user-images.githubusercontent.com/9912/132393570-e3c08c63-02d8-4cda-9985-d3a884510594.png">|

### > 1600px

| Before | After |
| :---: | :---: |
|<img width="1358" alt="CleanShot 2021-09-07 at 13 30 04@2x" src="https://user-images.githubusercontent.com/9912/132393648-b553da4f-ed86-49ff-bb1f-5150cea2b1bb.png">|<img width="1355" alt="CleanShot 2021-09-07 at 13 30 16@2x" src="https://user-images.githubusercontent.com/9912/132393676-c83b5152-b1bb-4384-ac86-7845586c9413.png">|

Fixes #110165.

smith committed:
Ensure name columns are not cut off on APM tables

Set a fixed width for the name column of each table and make the other elements overflow hidden.

Put the values to the left of the sparklines instead of the right.

This makes it so that at lower screen widths the metrics columns can be cut off, but the name column with alway be visible.

* Make the width of all name columns on the inventory and overview 256px
* Make values be shown before sparklines
* Make SparkPlots `overflow: 'hidden'`
* Make sparkplot and impact cells `width: 'auto'`
* Replace ServiceListMetric with SparkPlot so sparklines are not hidden on inventory

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/)
* Commit: 3878f1bf1c26b182eafd889f1e38a6e9c4e3882b
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-111046/3878f1bf1c26b182eafd889f1e38a6e9c4e3882b)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/execution/node/190/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/","number":"150757","commit":"3878f1bf1c26b182eafd889f1e38a6e9c4e3882b"}]}
PIPELINE-->

smith committed:
Merge remote-tracking branch 'upstream/master' into nls/table-width2

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/)
* Commit: 9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-111046/9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/execution/node/190/log)

### History
* :broken_heart: [Build #150757](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/) failed 3878f1bf1c26b182eafd889f1e38a6e9c4e3882b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/","number":"150757","commit":"3878f1bf1c26b182eafd889f1e38a6e9c4e3882b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/","number":"150807","commit":"9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3"}]}
PIPELINE-->

formgeist commented:
@smith I appreciate how difficult it is to build something responsive for all of these different panels and tables, so thanks for picking this up and coming up with a proposal. I'm not entirely sure how to best construct this in the code, but I feel it might be better to take the approach of possibly hiding away some of the "extra" information there is in the tables, like for instance the sparkplots. We did this for the service inventory. The sparkplots great if you have the required space, but I think as soon as we go down to those smaller viewports, we need to think more about what's most relevant to show and I still think the values are more important than the trends in this case. We did discuss other ways to display the sparkplots (possibly in a hidden popover or similar) but we haven't had a look at those possibilities yet, so really I'm just looking at how we can best and most easily resolve the issue we have on those breakpoints for smaller viewports.

![CleanShot 2021-09-03 at 11 34 49@2x](https://user-images.githubusercontent.com/4104278/131984477-60bd3b72-e29d-4f4f-ae6a-33a872e63439.png)

I think on a regular size it's now harder to read which column the sparkplot is related to. Also the name (link) columns are truncated so soon and they probably don't need to be for larger viewport sizes.

So I tried to mock how it might look like if the values were right-aligned going against the sparkplot. The main reason being that we got feedback that the values should be right-aligned for easier scannability like numbers typically are in tables.

<img width="1148" alt="CleanShot 2021-09-03 at 10 28 18@2x" src="https://user-images.githubusercontent.com/4104278/131984313-5fc2704f-af3e-4f35-9670-4a063f173af8.png">

I'm not sure we gain a lot on this for smaller viewport sizes, so I tried to mock how we might change certain flex properties and sizes to improve the layout as we go down the breakpoints. The main focus on two specific viewport sizes that I imagine are most common and where the left sidebar navigation is still shown (on lower sizes it collapses into a flyout).

The main changes in the mocks are;

- Changed the flex direction to columns for the side-by-side panels.
- Changed the height of the charts to align better with the Latency chart now that the panels are no longer side-by-side.
- Hide the sparkplot; I've hidden it just for the large instances table on the `XL` version, and all are hidden for the `L` version. Not sure if this level of specificity is what we want, but it's where I could see the potential of keeping as much of the content intact as possible.
- Added a fixed width to the impact indicator to `100px` for all sizes.

**EuiBreakpoint size XL**

![screencapture-localhost-5601-uyb-app-apm-services-opbeans-java-overview-2021-09-03-10_35_25](https://user-images.githubusercontent.com/4104278/131984903-b0a20fbc-872a-464e-81b6-93e870ae3fc5.png)

**EuiBreakpoint size L**

![screencapture-localhost-5601-uyb-app-apm-services-opbeans-java-overview-2021-09-03-10_38_39](https://user-images.githubusercontent.com/4104278/131984937-0b474fdd-4639-4b86-b887-c29513257697.png)

Thoughts?

smith committed:
updates based on feedback

smith commented:
@formgeist I've implemented some of your suggested changes in 7032bb963e6:

> Added a fixed width to the impact indicator to 100px for all sizes.

Done. It's 96 (`unit * 6`)

> the name (link) columns are truncated so soon and they probably don't need to be for larger viewport sizes.

Removed the fixed width for these so they are as wide as can fit.

> So I tried to mock how it might look like if the values were right-aligned going against the sparkplot.

Done.

A few questions on your comments:

> Changed the flex direction to columns for the side-by-side panels.

Does this mean no side-by-side panels at any size? If so this greatly simplifies things, since we're only seeing things get cut off in the side-by-side views for every table.

If we don't have any side-by-side panels, the only place where we have problems is on the instances table. This one has seven columns. We could hide the sparklines on this one when we're below XL, but even at the smallest value of XL (1200px) the values still get cut off a little bit:

<img width="1021" alt="CleanShot 2021-09-03 at 14 24 16@2x" src="https://user-images.githubusercontent.com/9912/132056555-f37076bf-f67d-4ba2-8c98-a2def1cea6a2.png">

I think if we:

* Get rid of side-by-side
* Hide sparklines at below XL on the instances table
* Be ok with the cutting off at 1200 in the screenshot above 

Then we'll be good. Please confirm if I'm understanding you correctly.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151066/)
* Commit: 7032bb963e6e0b8985fcde33f7f7df6e85f4ce8d
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151066/execution/node/417/log)

### History
* :broken_heart: [Build #150807](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/) failed 9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3
* :broken_heart: [Build #150757](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/) failed 3878f1bf1c26b182eafd889f1e38a6e9c4e3882b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/","number":"150757","commit":"3878f1bf1c26b182eafd889f1e38a6e9c4e3882b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/","number":"150807","commit":"9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151066/","number":"151066","commit":"7032bb963e6e0b8985fcde33f7f7df6e85f4ce8d"}]}
PIPELINE-->

formgeist commented:
> Does this mean no side-by-side panels at any size? If so this greatly simplifies things, since we're only seeing things get cut off in the side-by-side views for every table.

Sorry, that's not what I meant. I had created those mocks for breakpoints `XL` and `L` because 1200 px and below, the tables start to have issues with the sparklines, that even if we removed the sparkplots, the tables cells would still not show the appropriate information because of the content overflow. My suggestion is to go to 1 column layout at `XL` and below, since there's no longer enough space with the fixed left sidebar to show the side-by-side. Above 1200px the side-by-side works fine which include most laptop screensizes. 

> If we don't have any side-by-side panels, the only place where we have problems is on the instances table. This one has seven columns. We could hide the sparklines on this one when we're below XL, but even at the smallest value of XL (1200px) the values still get cut off a little bit:
> 
> <img alt="CleanShot 2021-09-03 at 14 24 16@2x" width="1021" src="https://user-images.githubusercontent.com/9912/132056555-f37076bf-f67d-4ba2-8c98-a2def1cea6a2.png">

If we hide the sparkplots at `XL` for the Instances table the values look OK to me;

<img width="905" alt="CleanShot 2021-09-06 at 15 57 53@2x" src="https://user-images.githubusercontent.com/4104278/132228411-ab910bf6-bb50-46a8-8fe6-c34dc9970431.png">

smith committed:
Merge remote-tracking branch 'upstream/master' into nls/table-width2

smith committed:
hide/show sparkplots

smith committed:
rename breakpoint to breakpoint

smith committed:
Merge remote-tracking branch 'upstream/master' into nls/table-width2

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

smith committed:
more fix

smith committed:
simplyfy listmetric

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151584/)
* Commit: f82ac43068ab87b4a7d375028f824e842467bf57
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Build OSS Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151584/execution/node/179/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151584/execution/node/197/log)
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151584/execution/node/190/log)

### History
* :broken_heart: [Build #151066](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151066/) failed 7032bb963e6e0b8985fcde33f7f7df6e85f4ce8d
* :broken_heart: [Build #150807](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/) failed 9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3
* :broken_heart: [Build #150757](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/) failed 3878f1bf1c26b182eafd889f1e38a6e9c4e3882b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/","number":"150757","commit":"3878f1bf1c26b182eafd889f1e38a6e9c4e3882b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/","number":"150807","commit":"9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151066/","number":"151066","commit":"7032bb963e6e0b8985fcde33f7f7df6e85f4ce8d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151584/","number":"151584","commit":"f82ac43068ab87b4a7d375028f824e842467bf57"}]}
PIPELINE-->

smith committed:
adjust chart height

smith committed:
Merge remote-tracking branch 'upstream/master' into nls/table-width2

andrewvc reviewed (approved):
Uptime changes LGTM

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151620/)
* Commit: dbddb3ac450222d6b8c504fa187b613a29ab1b36
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151620/execution/node/428/log)

### History
* :broken_heart: [Build #151584](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151584/) failed f82ac43068ab87b4a7d375028f824e842467bf57
* :broken_heart: [Build #151066](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151066/) failed 7032bb963e6e0b8985fcde33f7f7df6e85f4ce8d
* :broken_heart: [Build #150807](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/) failed 9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3
* :broken_heart: [Build #150757](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/) failed 3878f1bf1c26b182eafd889f1e38a6e9c4e3882b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150757/","number":"150757","commit":"3878f1bf1c26b182eafd889f1e38a6e9c4e3882b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/","number":"150807","commit":"9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151066/","number":"151066","commit":"7032bb963e6e0b8985fcde33f7f7df6e85f4ce8d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151584/","number":"151584","commit":"f82ac43068ab87b4a7d375028f824e842467bf57"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151620/","number":"151620","commit":"dbddb3ac450222d6b8c504fa187b613a29ab1b36"}]}
PIPELINE-->

cauemarcondes reviewed (approved):
LGTM

smith committed:
snapshot/import fix

formgeist commented:
@smith Thanks for the updates, it's starting to look really good!

I did find some things that perhaps were misunderstood from my feedback earlier. Also happy to sync later if that's easier?

### Breakpoint L

- The new columns look great, but I think we also want to hide all the sparkplots in this case except for the one in the Errors table which still works because there's only a single sparkplot.
![CleanShot 2021-09-08 at 09 47 35@2x](https://user-images.githubusercontent.com/4104278/132469397-d401b409-9c25-4bf4-997a-483d32b707db.png)

### Breakpoint XL

- We still show the panels in one column instead of side-by-side because in general, the space is too narrow for the main chart to be readable, it's just too squished.
![CleanShot 2021-09-08 at 09 45 22@2x](https://user-images.githubusercontent.com/4104278/132469698-a7dc3b2d-7316-4859-a406-07bbdaa1ad74.png)

### Above breakpoint XL (1200px+)

- I noticed that the sparkplots are all using the smaller option `compact` in the tables, but they should be using the default size because there's enough space for the larger screen sizes.

<img width="1682" alt="CleanShot 2021-09-08 at 09 35 39@2x" src="https://user-images.githubusercontent.com/4104278/132468950-257a97e4-7b47-4bc5-8dc9-282228b9ac02.png">

shahzad31 commented on x-pack/plugins/apm/public/hooks/use_breakpoints.ts:
```diff
@@ -24,17 +28,19 @@ export function getScreenSizes(windowWidth: number) {
 }
 
-export function useBreakPoints() {
+export function useBreakpoints() {
```
we should move this hook to observability plugin or better to kibana_react package.

smith replied:
Perhaps, but EUI has `useIsWithinBreakpoints` as well, and I find the return values from this hook confusing. I think we should rethink its API and see if the EUI team has suggestions before moving it.

formgeist reviewed (approved):
Design LGTM  👍

smith committed:
Merge remote-tracking branch 'upstream/master' into nls/table-width2

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151817/)
* Commit: 66e88da97cb2248fab016cfe0f88a0cf19ccdd38
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-111046/66e88da97cb2248fab016cfe0f88a0cf19ccdd38)
* [Documentation Changes](https://kibana_111046.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/178b56980f5744339cd7344d57ef529836a32172) | [after](https://github.com/elastic/kibana/commit/66e88da97cb2248fab016cfe0f88a0cf19ccdd38) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +2.3KB |

### History
* :green_heart: [Build #151670](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151670/) succeeded ed933e590c951f10c1b4953feb21c417b7d0918b
* :broken_heart: [Build #151620](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151620/) failed dbddb3ac450222d6b8c504fa187b613a29ab1b36
* :broken_heart: [Build #151584](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151584/) failed f82ac43068ab87b4a7d375028f824e842467bf57
* :broken_heart: [Build #151066](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151066/) failed 7032bb963e6e0b8985fcde33f7f7df6e85f4ce8d
* :broken_heart: [Build #150807](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/) failed 9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/150807/","number":"150807","commit":"9aa53d1c8f2135dd7a92dceabb5081fd4f2bf4c3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151066/","number":"151066","commit":"7032bb963e6e0b8985fcde33f7f7df6e85f4ce8d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151584/","number":"151584","commit":"f82ac43068ab87b4a7d375028f824e842467bf57"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151620/","number":"151620","commit":"dbddb3ac450222d6b8c504fa187b613a29ab1b36"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151670/","number":"151670","commit":"ed933e590c951f10c1b4953feb21c417b7d0918b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/151817/","number":"151817","commit":"66e88da97cb2248fab016cfe0f88a0cf19ccdd38"}]}
PIPELINE-->

smith commented:
Created #111577 with some ideas for further improvements.

kibanamachine commented:
The following labels were identified as gaps in your version labels and will be added automatically:
- v7.16.0

If any of these should not be on your pull request, please manually remove them.

kibanamachine commented:
## 💔 Backport failed

| Status | Branch | Result |
|:------:|:------:|:------:|
| ❌ |  7.15  | Commit could not be cherrypicked due to conflicts |
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/111578)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/111578">](https://github.com/elastic/kibana/pull/111578) |

Successful backport PRs will be merged automatically after passing CI.

To backport manually run:
`node scripts/backport --pr 111046`
